src/main/java/com/redislabs/redistimeseries/RedisTimeSeries.java

```diff
@@ -201,24 +169,27 @@ public boolean alter(String key, Map<String, String> labels) {
    */
   public boolean alter(String key, long retentionTime, Map<String, String> labels) {
     try (Jedis conn = getConnection()) {
+      byte[][] args = tsAlterArgs(key, retentionTime, labels);
+      return sendCommand(conn, Command.ALTER, args).getStatusCodeReply().equals("OK");
+    }
+  }
 
-      byte[][] args = new byte[3 + (labels==null ? 0 : 2*labels.size()+1)][];
-      int i=0;
-
-      args[i++] = SafeEncoder.encode(key);
+  private static byte[][] tsAlterArgs(String key, long retentionTime, Map<String, String> labels) {
+    byte[][] args = new byte[1 + (retentionTime>=0 ? 2 : 0 ) + (labels==null ? 0 : 2*labels.size()+1)][];
+    int i=0;
+    args[i++] = SafeEncoder.encode(key);
+    if (retentionTime>=0){
```

Comment:
this is a behavior change and will hide user's mistakes (giving negative numbers)